### PR TITLE
auto in function decl is invalid c++14

### PR DESCRIPTION
**Pull request by Karsten1987**
the use of `auto` in function declarations actually never made it officially into the cpp14 standard. 

http://stackoverflow.com/questions/30665506/auto-not-allowed-in-function-prototype-with-clang

### CODE REVIEW

File: cli_tools/src/rosnode_list.cpp [hunk 1]
```diff
@@ -31,14 +31,14 @@ print_usage()
   printf("-v: verbose output prints\n");
 }
 
-void print(auto & node_names)
+void print(std::vector<std::string> & node_names)
```
Comment (dirk-thomas) [inline cli_tools/src/rosnode_list.cpp]: The argument should be `const`.

Same below.

**Comment by Karsten1987** [comment]
fixed and rebased

**Comment by wjwwood** [comment]
I don't know what happened but the diff no longer includes the removal of `auto`. It only has the adding of `const` as if the `auto` was never there... Maybe I'm missing something. Also the diff line that @dirk-thomas commented on (https://github.com/ros2/cli_tools/pull/7#discussion_r105520133) is no longer there.

**Comment by wjwwood** [comment]
I think it is because @mikaelarguedas already removed them in https://github.com/ros2/cli_tools/commit/175d4a54681ebfc1c20d5e5637d8cd7e96c04660.

**Comment by Karsten1987** [comment]
yeah, there was something weird. I also didn't really know why I had to rebase again. But the file content is complete.
I'll merge it as is.

**Comment by mikaelarguedas** [comment]
yep my bad I forgot to make them const when I removed the use of auto :+1: for merging

**Pull request merged**